### ISSUE TITLE
running spin plugin update concurrently trigger git clone errors

### ISSUE THREAD

**Issue by rajatjindal** [fermyon/spin#997]
i am running some spin tests concurrently, and as part of it I run `spin plugin update`

As part of that, I am running into following error:

```
Error cloning Git repo https://github.com/fermyon/spin-plugins/: fatal: destination path '/home/runner/.local/share/spin/plugins/.spin-plugins' already exists and is not an empty directory.
```

While I understand this may be out of scope for supporting in `spin` as such, I wanted to create the issue here to collect your feedback on this.

**Comment by itowlson** [comment]
There seems to be a dearth of cross-platform cross-process synchronisation tools.  Windows has the mutex, but Unix seems to lack anything at the OS level.  There is a Rust crate `dir-lock` (https://docs.rs/dir-lock/latest/dir_lock/struct.DirLock.html) which uses a filesystem directory as a mutex, but it's a bit kludgy, and if cleanup failed then the user would be locked out of installing plugins, so I'm a bit wary of that.

Another possibility would be to do what templates do and do the git clone to a temporary directory, then copy the content to the actual directory.  This is still not robust against concurrency though.

**Comment by rajatjindal** [comment]
to be fair, i am running into this only because i am running e2e-tests which runs concurrently. 

I am not sure if an enduser would run the plugin install command concurrently (at least not until we reach 100's of plugins, and someone tries to setup all of them). 

i have implemented a workaround on tests side (by using file system based lock), so feel free to close the issue if you are inclined towards that.

**Comment by itowlson** [comment]
Fixed via #1569.  THE BADGER GIVETH
